lms/static/js/dashboard/views/course_view.js

```diff
@@ -0,0 +1,161 @@
+                        closeButton: ".close-modal"
+                    });
+
+                    $selector.attr('id', id);
+                    trigger = "#" + id;
+
+                    accessibleModal(
+                        trigger,
+                        _.sprintf("#%(type)s-modal .close-modal", {type: type}),
+                        _.sprintf("#%(type)s-modal", {type: type}),
+                        "#dashboard-main"
+                    );
+                },
+
+                toggleCourseActionsDropDown: function (e) {
+                    var ariaExpandedState,
+                        $currentTarget = this.$(e.currentTarget),
+                        index = $currentTarget.data('dashboard-index'),
+                    // Toggle the visibility control for the selected element and set the focus
+                        $dropDown = this.$('div#actions-dropdown-' + index);
+
+                    if ($dropDown.hasClass('is-visible')) {
+                        $dropDown.attr('tabindex', -1);
+                    } else {
+                        $dropDown.removeAttr('tabindex');
+                    }
+
+                    $dropDown.toggleClass('is-visible');
+
+                    // Inform the ARIA framework that the dropdown has been expanded
+                    ariaExpandedState = ($currentTarget.attr('aria-expanded') === 'true');
+                    $currentTarget.attr('aria-expanded', !ariaExpandedState);
+
+                    // Suppress the actual click event from the browser
+                    e.preventDefault();
+                },
+
+                unEnroll: function (e) {
+                    var $currentTarget = this.$(e.currentTarget),
+                        track_info = $currentTarget.data("track-info"),
+                        courseId = $currentTarget.data("course-id"),
+                        courseNumber = $currentTarget.data("course-number"),
+                        courseName = $currentTarget.data("course-name"),
+                        certNameLang = $currentTarget.data("cert-name-long"),
+                        refundInfo = $currentTarget.data("refund-info");
+
+                    $('#track-info').html(_.sprintf(track_info, {
+                        course_number: _.sprintf(
+                            "<span id='unenroll_course_number'>%(courseNumber)s</span>", {courseNumber: courseNumber}
+                        ),
+                        course_name: _.sprintf(
+                            "<span id='unenroll_course_name'>%(courseName)s</span>", {courseName: courseName}
+                        ),
+                        cert_name_long: _.sprintf(
+                            "<span id='unenroll_cert_name'>%(certNameLang)s</span>", {certNameLang: certNameLang}
+                        )
+                    }, true));
+
+                    $('#refund-info').html(refundInfo);
+                    $("#unenroll_course_id").val(courseId);
+                },
```

Comment:
This function does a lot of DOM manipulation, can't we just re-render this courses view with an updated model?